### ISSUE TITLE
Macro doesnt work above 250 characters

### ISSUE THREAD

**Issue by wing5wong** [Sellorio/mega-macro#224]
i have the following macro (253 characters). it does not function

```
#showtooltip
/cancelform [form:1/2/3/4]
/cancelaura soulshape
/leavevehicle [canexitvehicle]
/use [mod:shift] Grand Expedition Yak
/stopmacro [mod:shift]
/use [advflyable] Cliffside Wylderdrake
#/use Eve's Ghastly rider
/run C_MountJournal.SummonByID(0)
```

the macro functions once it is 250 characters.
for example, if i remove 3 characters from the `#/use Eve's Ghastly rider` line, the macro works as expected
```
#showtooltip
/cancelform [form:1/2/3/4]
/cancelaura soulshape
/leavevehicle [canexitvehicle]
/use [mod:shift] Grand Expedition Yak
/stopmacro [mod:shift]
/use [advflyable] Cliffside Wylderdrake
#/use Eve's Ghastly ri
/run C_MountJournal.SummonByID(0)
```

**Comment by bpapuni** [comment]
The addon reserves five characters for assigning Id's to the macros. Blizzard has removed the ability to have 1023 characters in a macro like we're used to, so now we are limited by the native 255 characters (minus the five MegaMacro reserves).

As a workaround, to save characters can you just use /cancelform without the conditional? You want to leave form regardless of what form you're in right?

**Comment by wing5wong** [comment]
This particular macro can definitely be optimised, but the issue will still exist for others.

I’d suggest in this case if 5 chars are reserved, the macro length should be shown as X/250 instead of 255 available


On 28 Jul 2024, at 4:26 PM, bpapuni ***@***.***> wrote:

﻿

The addon reserves five characters for assigning Id's to the macros. Blizzard has removed the ability to have 1023 characters in a macro like we're used to, so now we are limited by the native 255 characters (minus the five MegaMacro reserves).

As a workaround, to save characters can you just use /cancelform without the conditional? You want to leave form regardless of what form you're in right?

**Comment by Konungrr** [comment]
Having longer than 255 characters is the entire reason for the addon in the first place. So is this addon just outright dead?

**Comment by wing5wong** [comment]
Unless there are other macro addons to group by class/spec etc this is still useful for me

**Comment by Sellorio** [comment]
Yup exactly. The primary feature was always the macro swapping capability. It sucks that blizzard has removed this additional pillar but it is what it is.

The last thing I see as a pillar feature is the improved ui and editing capabilities.

**Comment by wing5wong** [comment]
Hi delirious
I still think this issue should remain open as the ui is giving misleading info. 255 available as opposed to 250

**Comment by Sellorio** [comment]
Fair enough

**Comment by wing5wong** [comment]
resolved by https://github.com/Sellorio/mega-macro/pull/225/commits/f15dbf924451a11220b0f78c7c382f396096248b
